Source/Entities/EntityInstanceBase.generated.cst

```diff
@@ -371,27 +375,29 @@ WriteRelationshipPropertyString();
             if (! cols[x].AllowDBNull && IsCSReferenceDataType(cols[x]))
             {
                 %>
-            ValidationRules.AddRule( CommonRules.NotNull,
-                new ValidationRuleArgs("<%=propertyName%>", "<%=friendlyName%>"));
+               ValidationRules.AddRule( CommonRules.NotNull,
+                   new ValidationRuleArgs("<%=propertyName%>", "<%=friendlyName%>"));
                 <%
             }
 
             if (CanCheckLength(cols[x]))
             {%>
-            ValidationRules.AddRule( CommonRules.StringMaxLength,
-                new CommonRules.MaxLengthRuleArgs("<%=propertyName%>", "<%=friendlyName%>", <%=getColumnSize(cols[x].Size)%>));
+               ValidationRules.AddRule( CommonRules.StringMaxLength,
+                   new CommonRules.MaxLengthRuleArgs("<%=propertyName%>", "<%=friendlyName%>", <%=getColumnSize(cols[x].Size)%>));
             <%
             }
 
             if (cols[x].DataType == DbType.Decimal)
             {%>
-            ValidationRules.AddRule( CommonRules.DecimalMaxLength,
-                new CommonRules.MaxLengthRuleArgs("<%=propertyName%>", "<%=friendlyName%>", <%=cols[x].Precision - cols[x].Scale%>));
+               ValidationRules.AddRule( CommonRules.DecimalMaxLength,
+                   new CommonRules.MaxLengthRuleArgs("<%=propertyName%>", "<%=friendlyName%>", <%=cols[x].Precision - cols[x].Scale%>));
             <%
             }
 
         }
         %>
+			   _commonRulesInitialised = true;

```

Comment:
I am not a fan of this initialisation check,  It is fairly crude.  It seems to work in all my tests, but I just wanted to flag it.

Reply:
Not thinking about the logic here, just looking at your tabbing changes, it looks like the tabs will not correctly formatting the resulting generated code.  An easier and cleaner solution from that perspective might be to write it like:

`if (_commonRulesInitialised)  return;`

Then you won't have another nested level of if statements.  And you won't have to worry about adjusting the generated tab/space level.

Reply:
Resolved